go-controller/pkg/node/gateway.go

```diff
@@ -169,16 +169,14 @@ func (g *gateway) Init(wf factory.NodeWatchFactory) error {
 	return nil
 }
 
-func (g *gateway) Run(stopChan <-chan struct{}, wg *sync.WaitGroup) {
+func (g *gateway) Start(stopChan <-chan struct{}, wg *sync.WaitGroup) {
 	if g.nodeIPManager != nil {
-		g.nodeIPManager.Run(stopChan)
```

Comment:
if we were to add a wg.Add(1) + defer here as well like below, will that have worked?